src/substrate/SubstrateContext.js

```diff
@@ -0,0 +1,65 @@
+import React, { useReducer } from 'react';
+import PropTypes from 'prop-types';
+import config from "../config";
+
+const INIT_STATE = {
+  socket: config.PROVIDER_SOCKET,
+  types: config.CUSTOM_TYPES,
+  keyring: null,
+  keyringState: null,
+  api: null,
+  apiState: null,
+};
+
+const reducer = (state, action) => {
+  // console.log("action dispatched", action, state);
```

Comment:
Debugging code. I'm fine leaving it if you think it might be helpful in the future, but if it was just a mistake, maybe remove it.